File: backend/java/Vehicle_Configurator/src/main/java/com/example/demo/services/ManufacturerService.java
package com.example.demo.services;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.example.demo.dto.ManufacturerDTO;
import com.example.demo.repositories.ManufacturerRepository;

@Service
public class ManufacturerService {

    @Autowired
    private ManufacturerRepository manufacturerRepository;

    public List<ManufacturerDTO> getManufacturersBySegment(int segmentId) {

        List<ManufacturerDTO> manufacturerList =
                manufacturerRepository.getManufacturersBySegment(segmentId);

        return manufacturerList;
    }

}

File: backend-dotnet/Services/ManufacturerService.cs
using backend_dotnet.DTO;
using backend_dotnet.Repository;

namespace backend_dotnet.Services
{
    public class ManufacturerService:IManufacturerService
    {

        private readonly IManufacturerRepository manufacturerRepository;


        public ManufacturerService(IManufacturerRepository manufacturerRepository)
        {
            this.manufacturerRepository = manufacturerRepository;
        }


        public List<ManufacturerDTO> GetManufacturersBySegment(int segmentId)
        {
            List<ManufacturerDTO> manufacturerList =
                manufacturerRepository.GetManufacturersBySegment(segmentId);


            return manufacturerList;
        }
    }
}
